src/List.elm

```diff
@@ -244,18 +238,20 @@ product numbers =
 
 {-| Find the maximum element in a non-empty list.
 
-    maximum [1,4,2] == 4
+    maximum 0 [1,4,2] == Just 4
+    maximum 0 []      == 0
 -}
-maximum : List comparable -> comparable
-maximum = foldl1 max
+maximum : comparable -> List comparable -> comparable
+maximum def list = foldl max def list
 
 
 {-| Find the minimum element in a non-empty list.
 
-    minimum [3,2,1] == 1
+    minimum 10 [3,2,1] == Just 1
+    minimum 10 []      == 10
 -}
-minimum : List comparable -> comparable
-minimum = foldl1 min
+minimum : comparable -> List comparable -> comparable
+minimum def list = foldl min def list
```

Comment:
Yeah I just realized today that there is a difference between a default
value (which will be ignored if the list is non-empty) and a base value for
foldl.

So I will change it back to maybe.

On Saturday, January 10, 2015, Evan Czaplicki <notifications@github.com>
wrote:

> In src/List.elm
> <https://github.com/elm-lang/core/pull/90#discussion-diff-22759619>:
>
> >  -}
> > -minimum : List comparable -> comparable
> > -minimum = foldl1 min
> > +minimum : comparable -> List comparable -> comparable
> > +minimum def list = foldl min def list
>
> I am having a hard time evaluating this change without any examples as
> data.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elm-lang/core/pull/90/files#r22759619>.
>